src/components/profiles-modals/CreateProfileModal.vue

```diff
@@ -0,0 +1,89 @@
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import { ModalCard } from "../all";
+import R2Error from "../../model/errors/R2Error";
+import sanitize from "sanitize-filename";
+
+@Component({
+    components: {ModalCard}
+})
+export default class CreateProfileModal extends Vue {
+    private newProfileName = '';
+    get isOpen(): boolean {
+        return this.$store.state.modals.isCreateProfileModalOpen;
+    }
+    closeModal() {
+        this.newProfileName = '';
+        this.$store.commit('closeCreateProfileModal');
+    }
+
+    get profileList(): string[] {
+        return this.$store.state.profiles.profileList;
+    }
+
+    makeProfileNameSafe(nameToSanitize: string): string {
+        return sanitize(nameToSanitize);
+    }
+
+    doesProfileExist(nameToCheck: string): boolean {
+        if ((nameToCheck.match(new RegExp('^([a-zA-Z0-9])(\\s|[a-zA-Z0-9]|_|-|[.])*$'))) === null) {
+            return true;
+        }
+        const safe: string = this.makeProfileNameSafe(nameToCheck);
+        return (this.profileList.some(function (profile: string) {
+            return profile.toLowerCase() === safe.toLowerCase()
+        }));
+    }
+
+    // User confirmed creation of a new profile with a name that didn't exist before.
+    // The profile can be either empty or populated via importing.
+    async createProfile() {
+        const safeName = this.makeProfileNameSafe(this.newProfileName);
+        if (safeName !== '') {
+            try {
+            document.dispatchEvent(new CustomEvent("created-profile", {detail: safeName}));
+            await this.$store.dispatch('profiles/addProfile', safeName);
+            this.closeModal();
+            } catch (e) {
+                const err = R2Error.fromThrownValue(e, 'Error whilst creating a profile');
+                this.$store.commit('error/handleError', err);
+            }
+        }
+    }
+}
+
+</script>
+<template>
+    <ModalCard v-if="isOpen" :is-active="isOpen" @close-modal="closeModal">
+
+        <template v-slot:header>
+            <p class="modal-card-title">Create a profile</p>
+        </template>
+
+        <template v-slot:body>
+            <p>This profile will store its own mods independently from other profiles.</p>
+            <input
```

Comment:
It would be nice for text input to be automatically selected so user can just start typing. I think we discussed this before the vacations but I don't remember if there was some problem with making it so. Not a blocker but possibly a low hanging UX improvement.

Reply:
autofocus is not working very well with vue, we _could_ probably do it with `refs`, but it gets a bit messy. probably would be best to figure out the best way to do it and apply it everywhere.